applications/Chat/coati/trainer/ppo.py

```diff
@@ -99,76 +118,20 @@ def __init__(self,
 
         self.device = get_current_device()
 
-    def _make_experience(self, inputs: Union[Tensor, Dict[str, Tensor]]) -> Experience:
-        if isinstance(inputs, Tensor):
-            return self.experience_maker.make_experience(inputs, **self.generate_kwargs)
-        elif isinstance(inputs, dict):
-            return self.experience_maker.make_experience(**inputs, **self.generate_kwargs)
-        else:
-            raise ValueError(f'Unsupported input type "{type(inputs)}"')
-
-    def _learn(self):
-        # replay buffer may be empty at first, we should rebuild at each training
-        if not self.sample_replay_buffer:
-            # HACK(cwher): according to the design of boost API, dataloader should also be boosted,
-            #  but it is impractical to adapt this pattern in RL training. Thus, I left dataloader unboosted.
-            dataloader = self.strategy.setup_dataloader(self.replay_buffer, self.dataloader_pin_memory)
-        if self.sample_replay_buffer:
-            pbar = tqdm(range(self.max_epochs), desc='Train epoch', disable=not is_rank_0())
-            for _ in pbar:
-                experience = self.replay_buffer.sample()
-                experience.to_device(self.device)
-                metrics = self.training_step(experience)
-                pbar.set_postfix(metrics)
+    def _make_experience(self, collect_step: int) -> Experience:
+        prompts = next(iter(self.prompt_dataloader))  # sample a prompt
```

Comment:
This usage is wrong and you should fix it.

Reply:
The original version wrote `prompts = next(iter(self.prompt_dataloader))`. [Link](https://github.com/hpcaitech/ColossalAI/blob/main/applications/Chat/coati/trainer/ppo.py) at line 153.

Reply:
Isn't this written on purpose? 

https://github.com/hpcaitech/ColossalAI/blob/95e95b6d588d27aeba8d62e6656392f771b69aee/applications/Chat/coati/trainer/ppo.py#L183-L186

The usage of `pretrain_dataloader` also follows this pattern.

Reply:
This is fixed in 128ec4f.